### PR TITLE
Update the casting logic of RMSNorm

### PR DESCRIPTION
**Pull request by lancerts**
## Summary
Fix the logic of RMSNorm, and isolate the casting logic from the computation logic.
If casting model =None, it does not perform casting but still performs the computation.

### CODE REVIEW

File: src/liger_kernel/ops/rms_norm.py [hunk 1]
```diff
@@ -104,7 +104,6 @@ def _rms_norm_backward_kernel(
     dW_ptr,
     dW_row_stride,
     n_cols,
-    eps,
```
Comment (lancerts) [inline src/liger_kernel/ops/rms_norm.py:107]: eps is not used in backward.

**Review by ByronHsu — COMMENTED** [review]
add a unit test to catch the bug?

File: src/liger_kernel/ops/rms_norm.py [hunk 2]
```diff
@@ -303,7 +302,6 @@ def backward(ctx, dY):
             X,
             W,
             r,
-            ctx.eps,
```
Comment (ByronHsu) [inline src/liger_kernel/ops/rms_norm.py:317]: does that mean we don't have to store eps in the forward pass?

**Comment by lancerts** [comment]
> add a unit test to catch the bug?


will do after we bump the pytorch to 2.3-2.4. It was added recently https://github.com/pytorch/pytorch/pull/121364 so 2.1, 2.2 cannot find the baseline torch implementation.

File: src/liger_kernel/ops/rms_norm.py (hunk 2 above)
Comment (lancerts) [reply src/liger_kernel/ops/rms_norm.py:317]: yes

**Comment by yundai424** [comment]
How about just adding a manual one by removing `.to(torch.float32)` from https://github.com/linkedin/Liger-Kernel/blob/d338f4b9923e452baecff6d36775242a5319df4c/test/transformers/test_rms_norm.py#L25 ?

**Review by ByronHsu — APPROVED** [review]

**Pull request merged**